metaflow/datastore/task_datastore.py

```diff
@@ -129,8 +149,15 @@ def __init__(
             self._objects = {}
             self._info = {}
         elif self._mode == "r":
-            if data_metadata is not None:
-                # We already loaded the data metadata so just use that
+            if data_metadata is not None and not any(
```

Comment:
logic to fetch metadata from S3 if the one from the metadata is incomplete. This way, we don't change metadata view at all. This is only used through client.